### ISSUE TITLE
It's impossible to trigger the `wrong_transmute` example

### ISSUE THREAD

**Issue by Dzejkop** [rust-lang/rust-clippy#9198]
### Description

The example is listed as follows:

```rust
let ptr: *const T = core::intrinsics::transmute('x')
```

however, it'll fail at compilation time, before even getting to Clippy.

[Proof](https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=a29264f0550437b3a01e2d6dedbebf7c)

In that case shouldn't this lint be deprecated?

### Version

```text
rustc 1.62.0 (a8314ef7d 2022-06-27)
binary: rustc
commit-hash: a8314ef7d0ec7b75c336af2c9857bfaf43002bfc
commit-date: 2022-06-27
host: x86_64-unknown-linux-gnu
release: 1.62.0
LLVM version: 14.0.5
```

**Comment by Jarcho** [comment]
The example does compile on targets with 32-bit pointers

**Comment by Dzejkop** [comment]
@Jarcho correct, but then it won't trigger the lint. Same if instead of `'x'` we use a `2u64` on 64-bit architectures.

As a matter of fact [here](https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=bdaad181d7e6f68ee864a91ca0c3b7f9) it'll trigger the `useless_transmute` lint instead.

**Comment by Jarcho** [comment]
It triggers fine when I tested it locally for a 32-bit target. It will also trigger on f32/f64->ref/ptr conversions.

`wrong_transmute` doesn't trigger on integers.

**Comment by Dzejkop** [comment]
Ok, yeah. Indeed it works for `f64`. Thanks!
